src/tbb/task.cpp

```diff
@@ -221,6 +221,31 @@ void notify_waiters(std::uintptr_t wait_ctx_addr) {
     governor::get_thread_data()->my_arena->get_waiting_threads_monitor().notify(is_related_wait_ctx);
 }
 
+d1::wait_tree_vertex_interface* get_thread_reference_vertex(d1::wait_tree_vertex_interface* wc) {
+    __TBB_ASSERT(wc, nullptr);
+    auto& dispatcher = *governor::get_thread_data()->my_task_dispatcher;
+
+    d1::reference_vertex* ref_counter{nullptr};
+    auto pos = dispatcher.m_reference_vertex_map.find(wc);
+    if (pos != dispatcher.m_reference_vertex_map.end()) {
+        ref_counter = pos->second;
+    } else {
+        if (dispatcher.m_reference_vertex_map.size() > 1000) {
+            for (auto it = dispatcher.m_reference_vertex_map.begin(); it != dispatcher.m_reference_vertex_map.end();) {
+                if (it->second->get_num_child() == 0) {
+                    it = dispatcher.m_reference_vertex_map.erase(it);
```

Comment:
I wanted to propose `std::erase_if` but realized it is a C++20 feature. Unfortunately:(